zingolib/src/wallet/keys/legacy.rs

```diff
@@ -81,7 +89,107 @@ pub(crate) fn legacy_fvks_to_ufvk<P: zcash_primitives::consensus::Parameters>(
         fvks.push(zcash_address::unified::Fvk::P2pkh(fvk_bytes));
     }
 
-    let ufvk = zcash_address::unified::Ufvk::try_from_items(fvks).map_err(|e| e.to_string())?;
+    let ufvk = zcash_address::unified::Ufvk::try_from_items(fvks)?;
 
     UnifiedFullViewingKey::decode(parameters, &ufvk.encode(&parameters.network_type()))
+        .map_err(|_| KeyError::KeyDecodingError)
+}
+
+pub(crate) fn legacy_sks_to_usk(
+    orchard_key: &orchard::keys::SpendingKey,
+    sapling_key: &sapling_crypto::zip32::ExtendedSpendingKey,
+    transparent_key: &extended_transparent::ExtendedPrivKey,
+) -> Result<UnifiedSpendingKey, KeyError> {
+    let mut usk_bytes = vec![];
+
+    // hard-coded Orchard Era ID due to `id()` being a private fn
+    usk_bytes.write_u32::<LittleEndian>(0xc2d6_d0b4)?;
+
+    CompactSize::write(
+        &mut usk_bytes,
+        usize::try_from(Typecode::Orchard).expect("typecode to usize should not fail"),
+    )?;
+    let orchard_key_bytes = orchard_key.to_bytes();
+    CompactSize::write(&mut usk_bytes, orchard_key_bytes.len())?;
+    usk_bytes.write_all(orchard_key_bytes)?;
+
+    CompactSize::write(
+        &mut usk_bytes,
+        usize::try_from(Typecode::Sapling).expect("typecode to usize should not fail"),
+    )?;
+    let sapling_key_bytes = sapling_key.to_bytes();
+    CompactSize::write(&mut usk_bytes, sapling_key_bytes.len())?;
+    usk_bytes.write_all(&sapling_key_bytes)?;
+
+    // the following code performs the same operations for calling `to_bytes()` on an AccountPrivKey in LRZ
```

Comment:
It's frustrating that we can't directly make use of the functions which have already written.